src/popup.tsx

```diff
@@ -49,187 +46,158 @@ import ContactSettings from "~routes/popup/settings/contacts/[address]";
 import NewContact from "~routes/popup/settings/contacts/new";
 import NotificationSettings from "~routes/popup/settings/notifications";
 import GenerateQR from "~routes/popup/settings/wallets/[address]/qr";
+import { ArConnectThemeProvider } from "~components/hardware/HardwareWalletTheme";
+import { AnimatePresence } from "framer-motion";
 
 export default function Popup() {
-  const theme = useTheme();
-  const [expanded, setExpanded] = useState(false);
-
-  // init popup
-  useSetUp();
+  const initialScreenType = useSetUp();
```

Comment:
The route guarding logic now lives here and doesn't rely on routes to work. That applies to the Unlock screen and an "undefined" state when the app first loads. This will also include a state and UI for the "creating wallet" state of the embedded wallet.